tests/matchers_unittest.py

```diff
@@ -18,75 +18,132 @@ def test_NotThisInt(self):
         self.assertEqual(testslide.matchers.NotThisInt(666), 42)
         self.assertNotEqual(testslide.matchers.NotThisInt(69), 69)
         self.assertNotEqual(testslide.matchers.NotThisInt(42), "derp")
+        with self.assertRaises(ValueError):
+            testslide.matchers.NotThisInt("derp")
+        with self.assertRaises(ValueError):
+            testslide.matchers.NotThisInt(1.340)
 
     def test_IntBetween(self):
         self.assertEqual(testslide.matchers.IntBetween(21, 666), 42)
         self.assertEqual(testslide.matchers.IntBetween(21, 666), 21)
         self.assertEqual(testslide.matchers.IntBetween(21, 666), 666)
         self.assertNotEqual(testslide.matchers.IntBetween(42, 69), 666)
         self.assertNotEqual(testslide.matchers.IntBetween(42, 69), "derp")
+        with self.assertRaises(ValueError):
+            testslide.matchers.IntBetween("derp", 42)
+        with self.assertRaises(ValueError):
+            testslide.matchers.IntBetween(42.42, "derp")
+        with self.assertRaises(ValueError):
+            testslide.matchers.IntBetween("derp", "derp")
 
     def test_IntGreaterThan(self):
         self.assertEqual(testslide.matchers.IntGreaterThan(21), 42)
         self.assertNotEqual(testslide.matchers.IntGreaterThan(21), 21)
         self.assertNotEqual(testslide.matchers.IntGreaterThan(21), 20)
         self.assertNotEqual(testslide.matchers.IntGreaterThan(42), "derp")
+        with self.assertRaises(ValueError):
+            testslide.matchers.IntGreaterThan("derp")
+        with self.assertRaises(ValueError):
+            testslide.matchers.IntGreaterThan(42.42)
```

Comment:
I would not make it fail in this case ...

The "concept" here is to have all the integer bigger than `42.42` so numbers like 43, 44, etc. I would say that constraining floats away from here would require users of the matchers to make more gymnastics with `floor`/`ceil`/`int` which might end up with tests that would be less obvious to be read

Reply:
Insightful, I really missed out the point that both `float` and `integer` values can be valid range extremes. I'll try to update accordingly as per the comments you mentioned. Thanks for your review. ☺